banyan/src/tree.rs

```diff
@@ -79,6 +82,46 @@ impl<
         }
     }
 
+    /// dumps the tree structure
+    pub fn dump_graph<S>(
```

Comment:
not sure about adding that as part of the public interface .. (also making `NodeInfo` public)

Reply:
would it be possible to implement `dot::GraphWalk` directly for `Tree` instead of creating an intermediate `ForestWithTree`?

Reply:
That's another option, and putting it behind a feature flag. The `dot` crate looks quite good to me, so that might be an option.

Reply:
@rklaehn WDYT?

Reply:
I would try to keep the number of deps of banyan itself small. Annoying newtype wrappers are a fact of life in rust.